core/src/main/java/org/elasticsearch/search/aggregations/AggregatorParsers.java

```diff
@@ -71,15 +71,14 @@ public AggregatorParsers(ParseFieldRegistry<Aggregator.Parser> aggregationParser
     /**
      * Parses the aggregation request recursively generating aggregator factories in turn.
      *
-     * @param parser    The input xcontent that will be parsed.
      * @param parseContext   The parse context.
      *
      * @return          The parsed aggregator factories.
      *
      * @throws IOException When parsing fails for unknown reasons.
      */
-    public AggregatorFactories.Builder parseAggregators(XContentParser parser, QueryParseContext parseContext) throws IOException {
-        return parseAggregators(parser, parseContext, 0);
+    public AggregatorFactories.Builder parseAggregators(QueryParseContext parseContext) throws IOException {
+        return parseAggregators(parseContext.parser(), parseContext, 0);
     }
 
     private AggregatorFactories.Builder parseAggregators(XContentParser parser, QueryParseContext parseContext, int level)
```

Comment:
isn't it enough to have the context as argument? then we can get the parser from the context as we do in other places?